src/hamster/workspace.py

```diff
@@ -0,0 +1,51 @@
+# - coding: utf-8 -
+
+# Copyright (C) 2015 Aurelien Naldi <aurelien.naldi at gmail.com>
+
+# This file is part of Project Hamster.
+
+# Project Hamster is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# Project Hamster is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with Project Hamster.  If not, see <http://www.gnu.org/licenses/>.
+
+from gi.repository import GObject as gobject
+from gi.repository import Wnck as wnck
+import dbus
+
+class WnckWatcher(gobject.GObject):
+    """
+    Listen for workspace changes with Wnck and forward them to the desktop integration agent.
+    It handles all wnck-specific parts and forwards standardised signals, based on the screensaver helper.
+    """
+    __gsignals__ = {
+        "workspace-changed": (gobject.SIGNAL_RUN_LAST, gobject.TYPE_NONE, (gobject.TYPE_PYOBJECT,gobject.TYPE_PYOBJECT))
+    }
+    def __init__(self):
+        gobject.GObject.__init__(self)
+
+        try:
+            self.bus = dbus.SessionBus()
+        except:
```

Comment:
As I'm not familiar with dbus, I copied this from idle.py, but all other places in hamster where a session bus is retrieved are unprotected, so I'm taking out the try/except here.